core/controllers/contributor_dashboard_admin.py

```diff
@@ -529,3 +530,215 @@ def _get_complete_translation_contribution_stats(
             })
 
         return response_translation_contribution_stats_dicts
+
+
+class ContributorDashboardAdminStatsHandlerNormalizedPayloadDict(TypedDict):
+    """Dict representation of ContributorDashboardAdminStatsHandler's
+    normalized_request dictionary.
+    """
+
+    page_size: int
+    offset: int
+    language_code: Optional[str]
+    sort_by: Optional[str]
+    topic_ids: Optional[List[str]]
+    num_days_since_last_activity: Optional[int]
+
+
+class ContributorDashboardAdminStatsHandler(
+    base.BaseHandler[
+        Dict[str, str],
+        ContributorDashboardAdminStatsHandlerNormalizedPayloadDict
+    ]
+):
+    """Return Contributor Admin Dashboard Stats for supplied parameters.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'contribution_type': {
+            'schema': {
+                'type': 'basestring'
+            },
+            'choices': [
+                feconf.CONTRIBUTION_TYPE_TRANSLATION,
+                feconf.CONTRIBUTION_TYPE_QUESTION
+            ]
+        },
+        'contribution_subtype': {
+            'schema': {
+                'type': 'basestring'
+            },
+            'choices': [
+                feconf.CONTRIBUTION_SUBTYPE_SUBMISSION,
+                feconf.CONTRIBUTION_SUBTYPE_REVIEW
+            ]
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'page_size': {
+                'schema': {
+                    'type': 'int'
+                },
+                'validators': [{
+                    'id': 'is_at_least',
+                    'min_value': 0
+                }],
+                'default_value': 20
+            },
+            'offset': {
+                'schema': {
+                    'type': 'int'
+                },
+                'validators': [{
+                    'id': 'is_at_least',
+                    'min_value': 0
+                }]
+            },
+            'language_code': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None,
+                'validators': [{
+                    'id': 'is_supported_audio_language_code'
+                }]
+            },
+            'sort_by': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None,
+                'choices': constants.CD_ADMIN_STATS_SORT_OPTIONS
+            },
+            'topic_ids': {
+                'schema': {
+                    'type': 'custom',
+                    'obj_type': 'JsonEncodedInString'
```

Comment:
Actually in the codebase where we have to pass list through a GET method we are using this technique. See for example **ConceptCardDataHandler**.

On trying the method you suggested the error still persist `{"error": "Schema validation for 'topic_ids' failed: Expected list, received topic1", "status_code": 400}`

Reply:
@kshitij01042002 Just a note, I imagine this discussion would have come up when the PR for ConceptCardDataHandler was created. Maybe find that PR using git's blame functionality and see if anything was mentioned there.

Reply:
The schema was introduced in this PR https://github.com/oppia/oppia/pull/14563 but there isn't any discussion about this.

Reply:
OK, I know what happened here -- @sagangwee is right.

JsonEncodedInString was introduced in https://github.com/oppia/oppia/commit/aefe95731d353fe7142f8b18343e49c81e39adcf because it was necessary to transfer an encoded object in a user-facing URL. 

Its use in data handlers was never actually intended, because you can transfer structured data through a JSON request and #14563 is actually wrong.

So I would suggest using the appropriate schema for a list of strings instead. We have a wiki page on schemas that you might find useful: https://github.com/oppia/oppia/wiki/Schemas

Reply:
Could you dig a little deeper into why it doesn't work? Propagating a bad pattern isn't really the way to go here.

Reply:
Hi @Nik-09 thanks for rolling in. I tried to dig more into it and found the similar conclusion. The URL query doesn't allow to pass the list but only string values. If we want to pass a list either we have to use the `POST` method or pass it as a `JsonEncodedInString`

@seanlip @sagangwee How should I proceed with it?

cc: @DubeySandeep

Reply:
Sorry, I amended my comment above -- I think what you have is probably fine.

Reply:
I think our validation logic doesn't support URL query params with the same key, though the `webapp2` framework does with `getall`: https://webapp2.readthedocs.io/en/latest/guide/request.html#get-data. So, what we have here seems fine.